### PR TITLE
Handle all filtering at the database layer

### PR DESCRIPTION
**Pull request by hxoliverrumbelow**
All filtering now gets converted to a MongoDB query and is executed on the database.

Running the test suite with debugging enabled will reveal the mongo queries:
```
$ reset && DEBUG=jsonApi:store:* npm test
```
```
jsonApi:store:mongodb search {"$and":[{"$or":[{"id":"cc5cca2e-0dd8-4b95-8cfc-a11230e73116"},{"id":"d850ea75-4427-4f81-8595-039990aeede5"},{"id":"32fb0105-acaa-4adb-9ec4-8b49633695e1"},{"id":"ad3aa89e-9c5b-4ac9-a652-6670f9f27587"}]},{"$or":[{"firstname":"Mark"}]}]}
jsonApi:store:mongodb search {"$and":[{"$or":[{"title":{"$lt":"M"}}]}]}
```

### CODE REVIEW

File: lib/mongoHandler.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ MongoStore._isRelationshipAttribute = function(attribute) {
 
 
 MongoStore._toMongoDocument = function(resource) {
-  var document = _.clone(resource, true);
```
Comment (DuncanFenning) [inline lib/mongoHandler.js]: We can 'un-require' clone if its been swapped to omit and isnt being used elsewhere

Comment (hxoliverrumbelow) [inline lib/mongoHandler.js]: Good call :+1:

**Comment by DuncanFenning** [comment]
Looks good :+1:

**Comment by connormeredith** [comment]
:+1:

**Pull request merged**